cpu/stm32l1/include/periph_cpu.h

```diff
@@ -98,6 +98,19 @@ typedef struct {
 #define PERIPH_SPI_NEEDS_TRANSFER_REGS
 /** @} */
 
+/**
+ * @brief   I2C device configuration
+ */
+typedef struct {
+    I2C_TypeDef *dev;       /**< pointer to the used I2C device */
+    gpio_t scl_pin;         /**< pin used for SCL */
+    gpio_t sda_pin;         /**< pin used for SDA */
+    uint8_t af_scl_sda;     /**< SCL or SDA AF */
+    uint8_t ev_irqn;        /**< I2C device Event Interrupt */
+    uint8_t er_irqn;        /**< I2C device Error Interrupt */
+    uint32_t clk_en;        /**< Clock enable register */
+} i2c_conf_t;
```

Comment:
I'm just talking without checking it: Do we really need all these information in `i2c_conf_t`? IIRC the UART type handles much less parameters. @thomaseichinger any suggestions?